Volume/Models/Article.swift

```diff
@@ -6,71 +6,86 @@
-        title: "Students Low On Cash Can Now Give TCAT Bus Drivers a Kiss On The Lips As Payment"
-    )
-
-private let cremeDeCornell =
-    Article(
-        date: Date(timeInterval: -20000, since: Date()),
-        image: "bulgogi",
-        isSaved: false,
-        publication: "Creme de Cornell",
-        shoutOuts: 1200,
-        title: "Vegan Bulgogi"
-    )
+struct Article: Codable, Hashable, Identifiable {
+    let articleURL: URL?
+    let date: Date
+    let id: String
+    let imageURL: URL?
+    let isSaved: Bool
+    let publication: Publication
+    let shoutOuts: Int
+    let title: String
+    
+    init(from article: GetHomeArticlesQuery.Data.Trending) {
+        articleURL = URL(string: article.articleUrl)
+        date = parse(date: article.date)
+        id = article.id
+        imageURL = URL(string: article.imageUrl)
+        isSaved = false // get from user defaults
+        publication = Publication(description: "", name: "", id: "", imageUrl: nil, isFollowing: false, recent: "")
+        shoutOuts = Int(article.shoutouts)
+        title = article.title
+    }
+    
+    init(from article: GetHomeArticlesQuery.Data.Following) {
+        articleURL = URL(string: article.articleUrl)
+        date = parse(date: article.date)
+        id = article.id
+        imageURL = URL(string: article.imageUrl)
+        isSaved = false // get from user defaults
+        publication = Publication(description: "", name: "", id: "", imageUrl: nil, isFollowing: false, recent: "")
+        shoutOuts = Int(article.shoutouts)
+        title = article.title
+    }
+    
+    init(from article: GetHomeArticlesQuery.Data.Other) {
+        articleURL = URL(string: article.articleUrl)
+        date = parse(date: article.date)
+        id = article.id
+        imageURL = URL(string: article.imageUrl)
+        isSaved = false // get from user defaults
+        publication = Publication(description: "", name: "", id: "", imageUrl: nil, isFollowing: false, recent: "")
+        shoutOuts = Int(article.shoutouts)
+        title = article.title
+    }
+}
 
-private let cuReview =
-    Article(
-        date: Date(timeInterval: -20, since: Date()),
-        image: nil,
-        isSaved: false,
-        publication: "Cornell Review",
-        shoutOuts: 0,
-        title: "The Cornell Student Body’s Problem with Tolerance"
-    )
+private func parse(date: String) -> Date {
+    let dateFormatter = ISO8601DateFormatter()
+    dateFormatter.formatOptions = [.withFullDate, .withTime, .withDashSeparatorInDate, .withColonSeparatorInTime]
+    return dateFormatter.date(from: date) ?? Date()
+}
 
-let articleData = [
-    cuReview, slopeMedia, cremeDeCornell, slopeMedia, cremeDeCornell,
-    slopeMedia, cuReview, cremeDeCornell, cuNooz, cremeDeCornell, cuNooz
-]
-// MARK: end of dummy data
+extension Array where Element == Article {
+    init(_ articles: [GetHomeArticlesQuery.Data.Trending]) {
+        self.init(articles.map(Article.init))
+    }
```

Comment:
Both here and in the struct, is there a way to combine all the functions with an enum since we're doing the same thing in each

Reply:
Unfortunately not... This is where I wish Swift was looser with type casting for objects with the same underlying structure.